### DISCUSSION TITLE
WASM Keyboard popup for mobile uX

### DISCUSSION THREAD

**Discussion by miketwenty1** [bevyengine/bevy#7045] [Q&A]
Is there a way to get the keyboard to popup when user input/text is needed?

If so, how?
If not, is there a bevy virtual keyboard available? or any other ideas to grab input from mobile users?

A crude way would be to create a bundle of keys and do my best at creating a makeshift keyboard on screen.. or putting in a javascript text field that populates a `#[wasm_bindgen]` function, but I'm trying to avoid these methods if possible.

**Comment by miketwenty1** [comment]
Example of makeshift on screen keyboard i'll use for the meantime if there's no better solution.
![example of makeshift screen keyboard](https://user-images.githubusercontent.com/7761473/209775183-639ba78b-19fb-4fe3-9643-8d9d6bcdcb4d.png)

**Comment by xietao3** [comment]
这是来自谢涛的自动回复邮件。
 
您好，我已经收到您的邮件。我将尽快给您回复。

**Comment by eblack-leaf** [comment]
[An offscreen text input](https://www.qt.io/blog/qt-for-webassembly-on-mobile-devices) seems to work at least until the [virtual keyboard api](https://developer.mozilla.org/en-US/docs/Web/API/VirtualKeyboard) stabilizes and it is included in web_sys as far as I can tell.
